pkg/vminetworkscontroller/vmi_controller.go

```diff
@@ -56,46 +56,42 @@ func (r *VirtualMachineInstanceReconciler) Reconcile(
 	defer cancel()
 	err := r.Client.Get(contextWithTimeout, request.NamespacedName, vmi)
 	if apierrors.IsNotFound(err) {
-		r.Log.Info("could not retrieve VMI - will cleanup its IPAMClaims")
-		if err := r.Cleanup(request.NamespacedName); err != nil {
-			return controllerruntime.Result{}, fmt.Errorf("error removing the IPAMClaims finalizer: %w", err)
-		}
-		return controllerruntime.Result{}, nil
+		vmi = nil
 	} else if err != nil {
 		return controllerruntime.Result{}, err
 	}
 
-	var ownerInfo metav1.OwnerReference
-	vm := &virtv1.VirtualMachine{}
-	contextWithTimeout, cancel = context.WithTimeout(ctx, time.Second)
-	defer cancel()
-	if err := r.Client.Get(contextWithTimeout, request.NamespacedName, vm); apierrors.IsNotFound(err) {
-		r.Log.Info("Corresponding VM not found", "vm", request.NamespacedName)
-		ownerInfo = metav1.OwnerReference{APIVersion: vmi.APIVersion, Kind: vmi.Kind, Name: vmi.Name, UID: vmi.UID}
-	} else if err == nil {
-		ownerInfo = metav1.OwnerReference{APIVersion: vm.APIVersion, Kind: vm.Kind, Name: vm.Name, UID: vm.UID}
-	} else {
-		return controllerruntime.Result{}, fmt.Errorf(
-			"error to get VMI %q corresponding VM: %w",
-			request.NamespacedName,
-			err,
-		)
+	vm, err := getOwningVM(ctx, r.Client, request.NamespacedName)
```

Comment:
thought about something
when getting owning VM, in case we still have the VMI we can validate it is the real owning VM according UID in the owner references,
and not a new VM that was created just after the original VM was deleted (with the same name)

if VMI == nil we can't assert this, but we can look on the ipamclaim ownerRef and compare the UID with the candid owning object (i think this what we should always do, both on VMI controller cases and
maybe also when checking realtionships in VM controller
https://github.com/kubevirt/ipam-extensions/pull/40/files#diff-dc5bb71846a22dd07f268340d8388c08689853df79bf3ae054514d77cecec888R56